hashintel/hash
# H-1181: Implement Inbox page for viewing notifications

## 🌟 What is the purpose of this PR?




This PR adds a the inbox page at `/inbox`, listing the current notifications of the user.

## 🔗 Related links





- H-1181

## 🔍 What does this change?




- adds an outgoing `occurredInBlock` link to all notifications
- renames the `occurredInEntity` link to `occurredInPage` link (it was decided to not account for mention notifications in quick notes, so we can be more specific with the name. If we do want to support mentions in things that aren't pages, we can rename this to `occurredInBlockCollection`)
- makes the after update/create hooks non-blocking in `createEntity` and `updateEntity` (i.e. we don't `await` their response)
- adds a check when creating notifications that ensures the recipient user can view the page where the notification was triggered.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?



The existing integration tests (slightly modified in this PR to account for additional changes) test the creation of the supported notifications.

There are no frontend tests.

## ❓ How to test this?



1. Checkout the branch locally and run HASH
2. Login as Bob, and create notifications for Alice by either:
   - Mentioning Alice in a page in the Example org
   - Mentioning Alice in a comment on a page in the Example org
   - Creating a comment on a block in a page created by Alice in the Example org
   - Replying to a comment left by Alice in a page in the Example org
3. Login as Alice, and view the inbox page at `/inbox` (or click on the link in the sidebar or the header)
4. Ensure the relevant notifications have been created
5. Click on "View Page/Comment" to go to the relevant block in the page where the notification was triggered
6. Click on "Mark as read" to mark the notification as read (it should disappear from the list)

## 📹 Demo



All currently supported notifications listed in the inbox page:

![image](https://github.com/hashintel/hash/assets/42802102/bae0869c-782f-4f16-989b-8e1bb163dce5)

The inbox page empty state:

<img width="1250" alt="image" src="https://github.com/hashintel/hash/assets/42802102/e54e17a2-68e6-40b3-98df-b67f06722143">
